Lib/test/support/__init__.py

```diff
@@ -2837,25 +2838,46 @@ def force_not_colorized(func):
     @functools.wraps(func)
     def wrapper(*args, **kwargs):
         import _colorize
-        original_fn = _colorize.can_colorize
-        variables: dict[str, str | None] = {
-            "PYTHON_COLORS": None, "FORCE_COLOR": None, "NO_COLOR": None
-        }
-        try:
-            for key in variables:
-                variables[key] = os.environ.pop(key, None)
-            os.environ["NO_COLOR"] = "1"
-            _colorize.can_colorize = lambda: False
+        from .os_helper import EnvironmentVarGuard
+
+        with (
+            swap_attr(_colorize, "can_colorize", lambda: False),
+            EnvironmentVarGuard() as env,
+        ):
+            for var in {"FORCE_COLOR", "NO_COLOR", "PYTHON_COLORS"}:
+                env.unset(var)
+            env.set("NO_COLOR", "1")
+
             return func(*args, **kwargs)
-        finally:
-            _colorize.can_colorize = original_fn
-            del os.environ["NO_COLOR"]
-            for key, value in variables.items():
-                if value is not None:
-                    os.environ[key] = value
+
     return wrapper
 
 
+def force_not_colorized_test_class(cls):
+    """Force the terminal not to be colorized for the entire test class."""
+    original_setUpClass = cls.setUpClass
+
+    @classmethod
+    @functools.wraps(cls.setUpClass)
+    def new_setUpClass(cls):
+        import _colorize
+        from .os_helper import EnvironmentVarGuard
+
+        cls.enterClassContext(
+            swap_attr(_colorize, "can_colorize", lambda: False)
+        )
+        env = cls.enterClassContext(EnvironmentVarGuard())
+        for var in {"FORCE_COLOR", "NO_COLOR", "PYTHON_COLORS"}:
+            env.unset(var)
+        env.set("NO_COLOR", "1")
```

Comment:
The code is basically duplicated with force_not_colorized(). Can you write a context manager and reuse it in these two functions?

Reply:
Good idea, updated!